### ISSUE TITLE
Consumer/Producer instance configuration options

### ISSUE THREAD

**Issue by crobinson42** [weyoss/redis-smq#66]
I might be misunderstanding the library but I think the configuration options are a singleton pattern. If this is the case then there is no solution (without a complex spawning of `child_process`) to connecting to multiple different Redis hosts in the same node.js process.

Example:
```
const consumer = new Consumer({ redis: { db: 15, host: 'localhost', port: '6300' } });
const consumer = new Consumer({ redis: { db: 7, host: 'localhost', port: '6379 } });
```

**Comment by weyoss** [comment]
From the docs [https://github.com/weyoss/redis-smq/blob/master/docs/configuration.md](https://github.com/weyoss/redis-smq/blob/master/docs/configuration.md):

> RedisSMQ accepts a one-time configuration setup which can take place, usually, during your application bootstrap.

Here is an example from the docs:

```javascript
// filename: ./examples/javascript/consumer.js
'use strict';
const config = require('./config');
const { Consumer, setConfiguration } = require('redis-smq');

// Applying system-wide configuration
// This setup should be done during your application bootstrap
// Throws an error if the configuration has been already set up
setConfiguration(config);

// ...
```
So RedisSMQ accepts a one-time configuration setup which is applied system wide.

>  there is no solution (without a complex spawning of child_process) to connecting to multiple different Redis hosts in the same node.js process.

No, it's not possible. Not "complex spawning of child_process" is needed. If you have an environment with different Redis hosts you should run separate consumers and producers for the given hosts with different configurations. 

If you are using a multi-master or a master/slave Redis setup then, RedisSMQ out-of-box does support it.

**Comment by crobinson42** [comment]
> So RedisSMQ accepts a one-time configuration setup which is applied system wide.

Yes, this is the singleton pattern. It should be documented in the README so folks are clear of this limitation when using the lib.
